Automattic/pocket-casts-android
# Onboarding: no longer show onboarding once it has been completed

| 📘 Project: #524  |
| --- |

## Description
This PR persists whether the user has completed the onboarding flow so that we no longer display it once they've completed it. 

The only time the user should see onboarding more than once is if they immediately close the app when the onboarding screen is first presented.

## Testing Instructions

To test this, turn on the onboarding feature flag in `base.gradle`.

Run through these various scenarios to confirm that onboarding is/is-not presented again after logging out of the app and restarting the app. Keep in mind that onboarding will only be presented a second time if the user is logged out, so when testing a scenario where the user has logged into an account during onboarding, we need to log out before restarting the app to verify that the onboarding flag is working properly (i.e., if the user remains logged in, then even without adding the onboarding flag the onboarding flow would not be presented again).

### Scenarios where Onboarding should be presented again
1. Swipe back from initial sign up or log in onboarding screen

### Scenarios where Onboarding should NOT be presented again
1. Tap "Not Now" on initial sign up or log in onboarding screen
2.  After successfully logging into a pre-existing account
3. Swipe back from the Recommendations screen after creating an account
4. Swipe back from the Welcome Screen
5. Tap "Find My Next Podcast" on the Welcome Screen
6. Tap "Done" on the Welcome Screen

## Checklist
 
#### I have tested any UI changes...
